### PR TITLE
Provide `AppUpdateHelpContact` based on the `Country`

### PR DESCRIPTION
**Pull request by msasikanth**
https://app.shortcut.com/simpledotorg/story/7222/implement-criticalappupdatedialog

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/appupdate/AppUpdateHelpContact.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package org.simple.clinic.appupdate
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+
+@JsonClass(generateAdapter = true)
+data class AppUpdateHelpContact(
+    @Json(name = "display_text")
+    val displayText: String,
+    val url: String
```
Comment (Honey14) [inline app/src/main/java/org/simple/clinic/appupdate/AppUpdateHelpContact.kt]: What is the url for?

File: app/src/main/java/org/simple/clinic/appupdate/AppUpdateModule.kt [hunk 2]
```diff
@@ -32,4 +37,19 @@ open class AppUpdateModule {
...
+  @Provides
+  fun providesAppUpdateHelpContactBasedOnCountry(
+      remoteConfigReader: ConfigReader,
+      moshi: Moshi,
+      country: Country
+  ): Optional<AppUpdateHelpContact> {
+    val type = Types.newParameterizedType(Map::class.java, String::class.java, AppUpdateHelpContact::class.java)
+    val configAdapter = moshi.adapter<Map<String, AppUpdateHelpContact>>(type)
+
+    val videoJson = remoteConfigReader.string("app_update_help_contact", "")
```
Comment (Honey14) [inline app/src/main/java/org/simple/clinic/appupdate/AppUpdateModule.kt:50]: Can we update the naming to something more descriptive? `videoJson` does not say much

**Review by Honey14 — CHANGES_REQUESTED** [review]
Looks good, left 2 comments PTAL

File: app/src/main/java/org/simple/clinic/appupdate/AppUpdateHelpContact.kt (hunk 1 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/appupdate/AppUpdateHelpContact.kt]: URL will be opened when the user clicks on the contact button.
https://simpledotorg.slack.com/archives/CFK2PHJ1L/p1644906753850759?thread_ts=1644229157.747019&cid=CFK2PHJ1L

File: app/src/main/java/org/simple/clinic/appupdate/AppUpdateModule.kt (hunk 2 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/appupdate/AppUpdateModule.kt:50]: Ah, right. I forgot to change it when copying the code over.

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,11 @@
 - Bump Sentry Gradle plugin to v2.1.5
 - Update translations: `am-ET`, `ta-LK`, `kn-IN`, `bn-BD`, `bn-IN`. `pa-IN`, `mr-IN`, `ta-IN`, `si-LK`, `sid-ET`, `te-IN`
 - Bump flipper to 0.134.0
+- Provide `AppUpdateHelpContact` based on the `Country`
+
+### Changes
+
+- [In Progress: 16 Feb 2021] Add stronger app update notifications
```
Comment (Honey14) [inline CHANGELOG.md:15]: We don't have notifications to update app yet, stronger does not make sense. Can we remove stronger and rename it to "Add app update reminders"

**Review by Honey14 — CHANGES_REQUESTED** [review]
One comment, ptal

File: CHANGELOG.md (hunk 3 above)
Comment (msasikanth) [reply CHANGELOG.md:15]: We already show an app update dialog, we consider that to be a reminder or a notification. That's why we are adding a stronger "notification", not just system notification, but this whole system.

Comment (Honey14) [reply CHANGELOG.md:15]: It is not just notifying users, but majorly reminding them to update, just the `Add stronger app update reminders` makes more sense. WDYT?

Comment (msasikanth) [reply CHANGELOG.md:15]: TBH I do think they are both same, but making the change and pushing it.

**Review by Honey14 — APPROVED** [review]
LGTM

**Pull request merged**